src/tools/gamemodelconverter/src/chunkfactory.cpp

```diff
@@ -0,0 +1,1837 @@
+        StringClass str;
+        str.Format("W3D_CHUNK_ANIMATION_CHANNEL Unknown Animation Channel Type %x", channel->Flags);
+        captainslog_warn((const char *)str);
+    }
+    AddData(data, "W3D_CHUNK_ANIMATION_CHANNEL", "W3dAnimChannelStruct", "", (void *)rawData);
+}
+
+void Write_W3D_CHUNK_ANIMATION_CHANNEL(ChunkSaveClass &csave, ChunkTreePtr &data)
+{
+    W3dAnimChannelStruct *channel = (W3dAnimChannelStruct *)data->data->info->value;
+    if (channel->Flags > ANIM_CHANNEL_VIS) {
+        StringClass str;
+        str.Format("W3D_CHUNK_ANIMATION_CHANNEL Unknown Animation Channel Type %x", channel->Flags);
+        captainslog_warn((const char *)str);
+    }
+    WriteChunkRaw(csave, W3D_CHUNK_ANIMATION_CHANNEL, channel, data->data->chunkSize);
+}
+
+READ_WRITE_CHUNK(W3D_CHUNK_ANIMATION_HEADER, W3dAnimHeaderStruct)
+
+void Read_W3D_CHUNK_BIT_CHANNEL(ChunkLoadClass &cload, ChunkTreePtr &data)
+{
+    char *rawData = ReadChunkRaw(cload);
+    W3dBitChannelStruct *channel = (W3dBitChannelStruct *)rawData;
+    captainslog_dbgassert(data->data->chunkSize == sizeof(W3dBitChannelStruct)
+                + ((channel->LastFrame - channel->FirstFrame) / 8) * sizeof(uint8_t), // 8 is the number of bits in a byte
+        "Chunk W3D_CHUNK_BIT_CHANNEL size does not match data size");
+    if (channel->Flags > BIT_CHANNEL_TIMECODED_VIS) {
+        StringClass str;
+        str.Format("W3D_CHUNK_BIT_CHANNEL Unknown Animation Channel Type %x", channel->Flags);
+        captainslog_warn((const char *)str);
+    }
+    AddData(data, "W3D_CHUNK_BIT_CHANNEL", "W3dBitChannelStruct[]", "", (void *)channel);
+}
+void Write_W3D_CHUNK_BIT_CHANNEL(ChunkSaveClass &csave, ChunkTreePtr &data)
+{
+    W3dBitChannelStruct *channel = (W3dBitChannelStruct *)data->data->info->value;
+    if (channel->Flags > BIT_CHANNEL_TIMECODED_VIS) {
+        StringClass str;
+        str.Format("W3D_CHUNK_BIT_CHANNEL Unknown Animation Channel Type %x", channel->Flags);
+        captainslog_warn((const char *)str);
+    }
+    WriteChunkRaw(csave, W3D_CHUNK_BIT_CHANNEL, channel, data->data->chunkSize);
+}
+
+void Read_W3D_CHUNK_BOX(ChunkLoadClass &cload, ChunkTreePtr &data)
+{
+    char *rawData = ReadChunkRaw(cload);
+    captainslog_dbgassert(data->data->chunkSize == sizeof(W3dBoxStruct), "Chunk W3D_CHUNK_BOX size does not match data size");
+    W3dBoxStruct *box = (W3dBoxStruct *)rawData;
+
+    if (box->Attributes & 4) {
+        StringClass str;
+        str.Format("W3D_CHUNK_BOX Unknown Attribute 0x00000004");
+        captainslog_warn((const char *)str);
+    }
+
+    if (box->Attributes & 8) {
+        StringClass str;
+        str.Format("W3D_CHUNK_BOX Unknown Attribute 0x00000008");
+        captainslog_warn((const char *)str);
+    }
+
+    if (box->Attributes & 0xFFFFFE00) {
+        StringClass str;
+        str.Format("W3D_CHUNK_BOX Unknown Attributes %x", box->Attributes & 0xFFFFFE00);
+        captainslog_warn((const char *)str);
```

Comment:
Is the formatted string is only used for the captainslog, then you can also simply write

```cpp
captainslog_warn("W3D_CHUNK_BOX Unknown Attributes %x", box->Attributes & 0xFFFFFE00);
```